src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -404,6 +407,89 @@ class device {
         return 0;
     }
 
+    /* new KDS which supported CU subdevice */
+    int parseCUSubdevStat() const
+    {
+        std::vector<std::string> kdsstat;
+        std::string errmsg;
+        uint32_t num_cu = 0;
+
+        pcidev::get_dev(m_idx)->sysfs_get("", "kds_stat", errmsg, kdsstat);
+        for (auto& line : kdsstat) {
+           std::smatch sm;
```

Comment:
Please use boost::tokenizer. You can refer to src\runtime_src\core\tools\xbmgmt2\flash\firmware_image.cpp for usage

Reply:
Maybe tokenizer is more suitable for predefine and well organize data with simple separator?

Currently, my sysfs node is human readable. It is not able to extract data with just a separator. For example, the kds_state node is like (I would need number of CU and CU usage from this file):
**root@xsjminm50:/proj/xsjhdstaff1/minm/XRTs/XRT_xocl/tests/xrt/perf_IOPS# cat /sys/bus/pci/devices/0000\:03\:00.1/kds_stat** 
Kernel Driver Scheduler(KDS)
CU to host interrupt capability: 1
Interrupt mode: ert
Configured: 0
Number of CUs: 3
  CU[0] usage(0) shared(1) refcnt(0) intr(disable)
  CU[1] usage(0) shared(1) refcnt(0) intr(disable)
  CU[2] usage(0) shared(1) refcnt(0) intr(disable)

The cu_info sysfs node (I need CU name and CU address):
**root@xsjminm50:/proj/xsjhdstaff1/minm/XRTs/XRT_xocl/tests/xrt/perf_IOPS# cat /sys/bus/pci/devices/0000\:03\:00.1/cu.u.40894464/cu_info** 
Kernel name: bandwidth1
Instance(CU) name: kernel_1
CU address: 0x1400000
CU index: 1
Protocol: CTRL_CHAIN
Interrupt cap: 1
Interrupt ID:  1
--- Arguments ---
Number of arguments: 4
arg name: output_r
  size: 8
  offset: 0x10
  direction: input
arg name: input_r
  size: 8
  offset: 0x1c
  direction: input
arg name: blocks
  size: 4
  offset: 0x28
  direction: input
arg name: reps
  size: 4
  offset: 0x30
  direction: input